### ISSUE TITLE
Fan stops running (dangerous).

### ISSUE THREAD

**Issue by mirage335** [Valkirie/HandheldCompanion#1169]
**Device manufacturer**
GPD

**Device model**
GPD Win Mini 2004

**Handheld Companion Version**
0.21.6.1

**Describe the bug**
Fan stops running upon switching from the external power to battery power configured performance profile, at least when connected to Thunderbolt/USB4 dock.

**To Reproduce**
Plug into external PCI-E Thunderbolt/USB4 dock with NVIDIA graphics card. Close lid.

Set two profiles. Maybe one 'Default', one 'Fast', both with software fan control.

Set one profile for when on battery power, the other when on external power.

Unplug external PCI-E Thunderbolt/USB4 dock with NVIDIA graphics card.

Fan stops, only starting again when switched to a different profile or 'Hardware' control. When switching back to the original profile in 'Software' control, the fan stops again.

**Expected behavior**
Fan continues on the alternative fan profile.

Obviously this is indeed dangerous for the health of the hardware. Also, there are still some intermittent cases of Handheld Companion going to 100% CPU . The new version of Handheld Companion may be causing some issues reconnecting to the PCI-E Thunderbolt/USB4 dock with NVIDIA graphics card.

I would much prefer to reduce the fan speed from the BIOS if that worked, or through a hardware mod.

**Comment by mirage335** [comment]
Latest release is still broken, fan can still stop when switching back to battery power.

**Comment by Valkirie** [comment]
@romracer anything comes to mind here ?

**Comment by romracer** [comment]
When the device is unplugged, the current power profile should be discarded.  When that happens, fan control is reverted to hardware.  @mirage335 do you hear any indication the fan is starting to spin at this moment (particularly if the device is warm enough that it should spin)?

When the new power profile is applied, the fan _should_ be set back to what the new profile requests (software here).  Perhaps the EC writes to toggle to hardware on discard and then re-enable software are fired off in too quick of succession?

**Comment by mirage335** [comment]
@romracer No. The fan *stops*. At other times, the fan quick spin is heard, during normal operations, often when using copy/paste, doing stuff, but when the fan stops, it just stops.

After resetting configuration somehow, and running Handheld Companion 0.21.7.0, I though I had overcome this. For a while it seemed to work. But when I tried unplugging and replugging my eGPU (and external power with that of course), it stopped the fan, again.

Could be related to PCIE link state power saving settings. That is one of the things I had changed recently - setting to maximum power savings prevents successfully reconnecting the eGPU (by the way, that is solely by Thunderbolt/USB-C, no direct PCIE, OcuLink, etc).

**Comment by mirage335** [comment]
And now yet again I am not able to reproduce this dangerous intermittent bug.
